src/Npgsql.GeoJSON/GeoJSONHandler.cs

```diff
@@ -124,48 +125,48 @@ static Exception AllOrNoneCoordiantesMustHaveZ(NpgsqlParameter? parameter, strin
 
         #region Read
 
-        public override ValueTask<GeoJSONObject> Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription? fieldDescription = null)
-            => ReadGeometry(buf, async);
+        public override ValueTask<GeoJSONObject> Read(NpgsqlReadBuffer buf, int len, bool async, CancellationToken cancellationToken, FieldDescription? fieldDescription = null)
+            => ReadGeometry(buf, async, cancellationToken);
 
-        async ValueTask<Point> INpgsqlTypeHandler<Point>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription? fieldDescription)
-            => (Point)await ReadGeometry(buf, async);
+        async ValueTask<Point> INpgsqlTypeHandler<Point>.Read(NpgsqlReadBuffer buf, int len, bool async, CancellationToken cancellationToken, FieldDescription? fieldDescription)
+            => (Point)await ReadGeometry(buf, async, cancellationToken);
```

Comment:
Assuming we change the parameter order as above:

```suggestion
            => (Point)await ReadGeometry(buf, async, cancellationToken: cancellationToken);
```

Reply:
> If so, this needs to be done across the whole codebase.

...but please leave it without a default value for internal and private methods.
Otherwise it's too easy to forget to pass it on.
I know Rider/ReSharper warn about this situation but we can't expect everyone to use the tools we use.

Reply:
@Brar I think this should be taken care of by [CA2016](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2016?view=vs-2019) - this isn't a Rider/Resharper thing, it's a compiler-flagged thing which should fail the build. This is better than us making the code more verbose and explicitly passing `default` when not needed. 

(BTW note also [CA1068](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1068?view=vs-2019) about the token coming last)

Reply:
As I know Microsoft violated this rule in .NET internals many times.

Reply:
I don't follow... If we have this as an error, the build will fail - unless we choose to suppress it, no?

I'm guessing Microsoft suppressed/violated this because in some cases adding a default to an existing API may be complicated/breaking. Also, why should we care what Microsoft did?

Reply:
Am just trying to say that not following the rule is wrong. In some cases it's plausible.

Reply:
Can you point to an example?

Reply:
Not willing to continue that discussion 'cause want to spend my time on more constructive things.

Reply:
> I think this should be taken care of by CA2016
[...]
note also CA1068 about the token coming last

I didn't know that there's a standard for this, but if there is, I'm all for following it.

A question still is, why the compiler doesn't actually enforce this standard (or at least didn't when I did my initial experiments with token passing).
I initially had the default token on internal methods, but that led to several places across the codebase where I in fact forgot to pass the token along, wondering every single time why cancellation didn't work as expected.
The change to non-default and my opinion didn't come out of the nowhere.
Anyways, if there's a standard and we get a reliable warning from the compiler, let's follow the standard.
If the discussion boils down to standard vs. aesthetics I'm all for the standard.

Reply:
Well, it's not only about aesthetics, but also about intent. If a method doesn't support the default cancellation token, you have to pass it yourself - it's assured by the compiler.

Reply:
You're right, but then again the only way to prevent that somebody does pass you the default token would be to start your method with `if (cancellationToken == default) throw new ArgumentException("You can't pass the default cancellation token here"; nameof(cancellationToken ))`.
The signature doesn't really prevent it.

Reply:
It's not about prevention, but a difference between:

- This method requires a cancellation token, but I pass the default one because I know what I'm doing
- This method has an optional cancellation token, and I just ignore it

Reply:
@Brar you're absolutely right - CA2016 is completely new and being introduced in .NET 5.0, and it also seems to be opt-in.

Take a look at https://github.com/roji/Test/tree/NetAnalyzers for an example of the error in action.

Reply:
@vonzshik I think I don't understand your point ...

Technically, no method *requires* a cancellation token - it's always possible to provide CancellationToken.None. It's true that in a given codebase (like Npgsql), the method may always get *called* with a non-None cancellation token, but that doesn't mean the method should declare a non-optional token...

Bottom line, if we leave the token optional, is there any *problem* here that CA2016 doesn't solve for us?

Reply:
I don't think there should be any problem, and CA2016 together with CA1068 should be enough.